internal/proxy/proxy_test.go

```diff
@@ -83,12 +89,15 @@ func TestProxy(t *testing.T) {
 	})
 
 	t.Run("happy path with vault client", func(t *testing.T) {
+		vaultRequests = []*http.Request{}
 		fakeVaultResponse = vaultResponseFooBar
 		proxyVaultClient, err := api.NewClient(&api.Config{
 			Address: "http://" + proxyAddr,
 		})
 		require.NoError(t, err)
 		resp, err := proxyVaultClient.Logical().Read("secret/data/foo")
+
+		require.NotEmpty(t, vaultRequests[0].Header.Get("User-Agent"))
```

Comment:
Nice! Should we also assert that it contains "requesting from proxy"? And now that I am thinking about it, maybe do you think we should add a test to assert the other case where the user agent contains info about writing to the temp file?

Reply:
we should - for some reason i assumed we wouldn't know precisely what was in the user agent header, but obviously we do.